WalletWasabi/Services/SingleInstanceChecker.cs

```diff
@@ -37,22 +44,38 @@ public SingleInstanceChecker(int port, int timeoutMultiplier = 1)
 		Port = port;
 		_timeoutMultiplier = timeoutMultiplier;
 	}
+	
+	public event EventHandler? OtherInstanceStarted;
 
 	private int Port { get; }
 
 	private CancellationTokenSource DisposeCts { get; } = new();
 	private TaskCompletionSource? TaskStartTcpListener { get; set; }
 
-	public event EventHandler? OtherInstanceStarted;
-
+	public async Task<WasabiInstanceStatus> CheckSingleInstanceAsync()
+	{
+		// Start single instance checker that is active over the lifetime of the application.
+		try
+		{
+			var singleInstanceResult = await EnsureSingleOrThrowAsync().ConfigureAwait(false);
+			return singleInstanceResult
+				? WasabiInstanceStatus.NoOtherInstanceIsRunning
+				: WasabiInstanceStatus.AnotherInstanceIsRunning;
+		}
+		catch (Exception e)
+		{
+			Logger.LogError(e);
+			return WasabiInstanceStatus.PortIsBeingUser;
+		}
+	}
+	
 	/// <summary>
 	/// This function ensures that this is the only instance running on this machine or throws an exception if it is not. In case of secondary start
 	/// we try to signal the first instance before throwing the exception.
 	/// On macOS this function will never throw if you run Wasabi as a macApp, because mac prevents running the same APP multiple times on OS level.
 	/// </summary>
 	/// <exception cref="InvalidOperationException">Wasabi is already running, signaling the first instance failed.</exception>
```

Comment:
I understand these points and it seems that I must rename it because I changed the signature of the method but I don't want to fix bugs or improve anything in any way, if it would have possible to do my task without having to deal with this thing I would have done it.